app/root.tsx

```diff
@@ -30,8 +30,13 @@ export const links: LinksFunction = () => [
 ];
 
 export async function loader({ request, context }: LoaderArgs) {
+  const effectiveUsername = await getEffectiveName(request);
+  const userBeforeEffectiveName = await getUser(request);
```

Comment:
Question: if both `getEffectiveName` and `getUser` both require `request`, can you not return the `effectiveUsername` on the result of `getUser` (i.e., you could do this internally in `getUser`)?